CareKit/CareKitTests/TestAnyEventStoreExtensions.swift

```diff
@@ -118,6 +118,10 @@ class TestAnyEventStoreExtensions: XCTestCase {
             didUpdate.fulfill()
         }
 
+        #if compiler(>=5.8.0)
+        await fulfillment(of: [didUpdate], timeout: 2)
+        #else
         wait(for: [didUpdate], timeout: 2)
```

Comment:
This is a warning in Swift 5.8.0 that will become an error in Swift 6.0+

Reply:
Good catch with this, a similar change is needed in CareKitStore/TestCoreDataQueryMonitor.swift in testIrrelevantChangeDoesNotProduceResult. I do wonder if `await withCheckedContinuation` might be more suitable but that'll require significant changes to both tests.
<img width="617" alt="Screenshot 2023-04-08 at 11 27 55 pm" src="https://user-images.githubusercontent.com/1451896/230745612-e01e15b0-5157-4c0b-a803-5813e711679f.png">

Reply:
I don't believe that will work as `wait` is known to cause deadlocks in XCTest and `fulfillment` was introduced to solve the issue. Some details are in [Xcode 14.3 changelog](https://developer.apple.com/documentation/xcode-release-notes/xcode-14_3-release-notes)